demo/vl-map-wms-layer.html

```diff
@@ -39,6 +39,38 @@
         </vl-map-image-wms-layer>
       </vl-map>
     </vl-demo>
+
+    <vl-demo data-vl-title="Kaart met Image WMS kaartlaag met custom SLD">
+      <vl-map id="map-with-image-wms-layer">
+        <vl-map-baselayer-grb-gray></vl-map-baselayer-grb-gray>
+        <vl-map-image-wms-layer
+          data-vl-name="Overstromingsgevaarkaarten"
+          data-vl-version="1.1.0"
+          data-vl-opacity="0.6"
+          data-vl-url="https://geoservice.waterinfo.be/Overstromingsgevaarkaarten-PLUVIAAL/wms"
+          data-vl-layers="Overstromingsgevaarkaarten-PLUVIAAL:overstroombaar_gebied_PLU_noCC"
+          data-vl-sld-body='<StyledLayerDescriptor xmlns="http://www.opengis.net/sld" xmlns:ogc="http://www.opengis.net/ogc" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.0.0" xsi:schemaLocation="http://www.opengis.net/sld StyledLayerDescriptor.xsd">
```

Comment:
Da lijkt precies wel wa veel om via attribuut te setten?
Een slot is misschien niet echt praktischer omdat ge dan xml en html gaat mengen, echter, en een setter is niet per se praktischer dan een attribuut. Heb dus niet echt een beter voorstel...

Reply:
Misschien een `vl-map-styled-layer-descriptor` component maken die het stylen abstraheert en een `toStyledLayerDescriptor()` functie heeft dat uiteindelijk gebruikt wordt door de wms-layer?

Reply:
Ik zie voorlopig niet bijzonder veel meerwaarde om het te wrappen in extra component. Heb ook nagedacht om het in de vl-map-layer-style te integreren, maar denk dat ons ook wel wat ver zou brengen. Dit is ook meer specifiek voor WMS om het op server side te stijlen.

Reply:
In principe komt dat "qua API" wel meer overeen met hoe het bij WFS in onze abstractie in mekaar steekt.

Reply:
Daar is het dan de style die 'zichzelf zet' op de layer ...

Reply:
Maar dan blijft ge toch ook zitten met het feit of die style component zelf zijn SLD binnen krijgt als data, of die zelf maakt (als het telkens een subclass is) of toch verder onderverdeeld is in componentjes volgens die XML hier ...

Reply:
Dat lijkt me wel beter om een beetje gelijk te trekken met de features layer style. Ik maak dan een component <vl-map-wms-style> met hierin attribuut data-vl-sld (waar dan de XML kan komen), ok?

Reply:
Blijkbaar werkt dit toch ook niet zoals het hoort (afnemende applicatie krijgt undefined op de _sldBody, precies dat die nog niet defined is), zelfs met de whenDefined, nog niet direct gevonden waarom niet

Reply:
@Pieter-Beckers staat je poging ergens op een branch beschikbaar zodat ik ook eens kan kijken naar wat er fout loopt?

Reply:
Nee, ik zal proberen het te reproduceren in een demo

Reply:
@coemans : de laatste commit bevat een demo die het probleem aantoont (demo/vl-map-wms-layer-dist.html). De SLD lijkt niet opgepikt te worden als ik de dist versie gebruik (kaart toont standaard stijl, en niet de paarse stijl zoals in de demo/vl-map-wms-layer.html ). Nog even gekeken naar het probleem, maar niet gevonden, kijk jij er nog even naar?

Reply:
Ik heb toch nog iets moeten wijzigen: de ready van de layer werd soms al gezet voor dat de _layer of _source werd gezet op de wms-layer, en dit klopte niet. Ik heb de asynchroniteit wat verbeterd. Kijk je nog even na?

Reply:
Ik heb nog een private functie call verwijderd, verder geen opmerkingen. Alleen de build zouden we terug groen moeten krijgen om merge en release te kunnen uitvoeren.